core/internal/cltest/factories.go

```diff
@@ -616,6 +616,29 @@ func MustInsertPipelineRun(t *testing.T, db *sqlx.DB) (run pipeline.Run) {
 	return run
 }
 
+func MustInsertPipelineRunWithStatus(t *testing.T, db *sqlx.DB, pipelineSpecID int32, status pipeline.RunStatus) (run pipeline.Run) {
+	var finishedAt *time.Time
+	var outputs pipeline.JSONSerializable
+	var allErrors pipeline.RunErrors
+	var fatalErrors pipeline.RunErrors
+	now := time.Now()
+	switch status {
+	case pipeline.RunStatusCompleted:
+		finishedAt = &now
+		outputs = pipeline.JSONSerializable{
+			Val:   "foo",
+			Valid: true,
+		}
+	case pipeline.RunStatusErrored:
+		finishedAt = &now
+		allErrors = []null.String{null.StringFrom("oh no!")}
+		fatalErrors = []null.String{null.StringFrom("oh no!")}
+	default:
```

Comment:
Is default valid here? or could it help debugging to blow up the test? e.g.
```suggestion
	default:
		t.Fatalf("unrecognized status %s", status)
```

Reply:
Its ok to do nothing here. Will make a comment